inginious/frontend/user_manager.py

```diff
@@ -1034,9 +1052,31 @@ def generate_api_key(cls):
         return hexlify(os.urandom(40)).decode('utf-8')
 
     @classmethod
-    def hash_password(cls, content):
+    def hash_password_sha512(cls, content):
         """
         :param content: a str input
         :return a hash of str input
         """
         return hashlib.sha512(content.encode("utf-8")).hexdigest()
+
+    @classmethod
+    def hash_password_argon2id(cls, content):
+        """
+        :param content: a str input
+        :return a hash of str input
+        """
+        ph = PasswordHasher()
+        return "argon2id-" + ph.hash(content)
+
+    @classmethod
+    def hash_password(cls, content):
```

Comment:
A potential implementation could be `hash_password(cls, content)`  that indeed use the latest method, with an additional `verify_hash(cls, db_hash, password, method="sha512")` that takes the method